UnitTests/CardTests/CardClient_Tests.swift

```diff
@@ -28,68 +30,247 @@ class CardClient_Tests: XCTestCase {
         cardClient = CardClient(config: config, apiClient: apiClient)
     }
 
+    override func tearDown() {
+        super.tearDown()
+        mockURLSession.clear()
+    }
+
     // MARK: - approveOrder() tests
 
-    func testApproveOrder_withValidJSONRequest_returnsOrderData() async {
-        let jsonResponse = """
-        {
-            "id": "testOrderID",
-            "status": "APPROVED",
-            "payment_source": {
-                "card": {
-                    "last_digits": "7321",
-                    "brand": "VISA",
-                    "type": "CREDIT"
-                }
-            }
-        }
-        """
+    func testApproveOrder_withNoThreeDSecure_returnsOrderData() {
+        if let jsonResponse = CardResponses.confirmPaymentSourceJson.rawValue.data(using: String.Encoding.utf8) {
+            let mockResponse = MockResponse.success(
+                MockResponse.Success(data: jsonResponse, urlResponse: successURLResponse)
+            )
+            mockURLSession.addResponse(mockResponse)
+            let expectation = expectation(description: "testName")
+
+            let cardRequest = CardRequest(orderId: "testOrderId", card: card)
+
+            let mockCardDelegate = MockCardDelegate(success: {_, result -> Void in
+                XCTAssertEqual(result.orderID, "testOrderId")
+                XCTAssertEqual(result.paymentSource?.card.brand, "VISA")
+                XCTAssertEqual(result.paymentSource?.card.lastFourDigits, "7321")
+                XCTAssertEqual(result.paymentSource?.card.type, "CREDIT")
+                expectation.fulfill()
+            }, error: { _, _ -> Void in
+                XCTFail()
+            }, threeDSWillLaunch: { _ -> Void in
+                XCTFail()
+            })
 
-        mockURLSession.cannedURLResponse = successURLResponse
-        mockURLSession.cannedJSONData = jsonResponse
+            cardClient.delegate = mockCardDelegate
 
-        let cardRequest = CardRequest(orderID: "", card: card)
+            cardClient.start(
+                request: cardRequest,
+                context: MockViewController(),
+                webAuthenticationSession: MockWebAuthenticationSession()
+            )
 
-        do {
-            let cardResult = try await cardClient.approveOrder(request: cardRequest)
-            XCTAssertEqual(cardResult.orderID, "testOrderID")
-            XCTAssertEqual(cardResult.brand, "VISA")
-            XCTAssertEqual(cardResult.lastFourDigits, "7321")
-            XCTAssertEqual(cardResult.type, "CREDIT")
-        } catch {
-            XCTFail()
+            waitForExpectations(timeout: 10)
+        } else {
+            XCTFail("Data json cannot be null")
         }
     }
 
-    func testApproveOrder_withInvalidJSONResponse_returnsParseError() async {
-        let jsonResponse = """
+    func testApproveOrder_whenApiCallFails_returnsError() {
+        if let jsonResponse = """
```

Comment:
+1 here, removing `if let` will help us remove conditional logic from the unit tests.